### PR TITLE
fix roles

### LINKED ISSUE #1794: User access rights with pending invitation
How is it possible that a User (user with Role  Atlantis01 NC 2020, and with pending invitation for Atlantis01 NC 2025) has access to 2025 Atlantis01 ?

### LINKED ISSUE #1795: User with one role in 2020: error accessing Global
Logged with a user with one role only (Atlantis01 2020 NC - https://fra-data.review.fao.org/X01/assessments/fra/2020/home/users/600) 

When accessing Global (https://fra-data.review.fao.org/WO/assessments/fra/2020/home/overview), it gives error userNotAuthorized

Image

### LINKED ISSUE #1801: User editing him self not authorized
User https://fra-data.review.fao.org/X01/assessments/fra/2020/home/users/600 tried to edit himself, but system gives not authorized error

### LINKED ISSUE #1802: Country Selector: user multiple cycles same country -> double entry
UserA is NC for Atlantis 01 in 2020 and 2025.

Image

to fix: show only cycle related countries

### PR DESCRIPTION
**Pull request by sorja**
Note: Roles are always cycle related (excl. Admin)

Resolves many
Resolves #1795 ✅
Resolves #1794 ✅
Resolves #1801 ✅
Resolves #1802 ✅

Role check adds cycle in this pr

- you dont have a role if you dont accept it

[Separate country list by roles](https://github.com/openforis/fra-platform/pull/1796/commits/bb5ffed6dfe771c2109bf6a593f9fcb868877f9b)
[bb5ffed](https://github.com/openforis/fra-platform/pull/1796/commits/bb5ffed6dfe771c2109bf6a593f9fcb868877f9b)

https://user-images.githubusercontent.com/5508251/205650413-12d76b66-c1e8-45dc-a652-10d2abcec307.mov

## Note: We should explore solution to add roles under cycleUuid

`roles: { [cycleUuid]: Role[] }`

### CODE REVIEW

**Review by minotogna — APPROVED** [review]

**Review by minotogna — APPROVED** [review]

**Comment by sorja** [comment]
pending changes

File: src/server/controller/user/invite.ts [hunk 1]
```diff
@@ -40,6 +40,7 @@ export const invite = async (
       t
     )
 
+    // TODO
```
Comment (minotogna) [inline src/server/controller/user/invite.ts]: What's here todo ?

Comment (sorja) [reply src/server/controller/user/invite.ts]: @rfontanarosa

Comment (minotogna) [inline src/server/controller/user/invite.ts]: remove this TODO

**Review by minotogna — CHANGES_REQUESTED** [review]

**Review by rfontanarosa — APPROVED** [review]

File: src/client/components/CountrySelect/CountryList/CountryList.tsx [hunk 2]
```diff
@@ -67,8 +68,17 @@ const CountryList: React.FC<Props> = (props: Props) => {
           })}
         </div>
 
-        {Object.entries(countryMap).map(([role, countries]) => {
-          return <CountryListRoleSection key={role} role={role} countryISOs={countries} query={query} />
+        {Object.entries(countryMap).map(([role, cycleCountries]) => {
+          const countries = cycleCountries[cycle.uuid]
+          return (
+            <CountryListRoleSection
+              cycleUuid={cycle.uuid}
```
Comment (minotogna) [inline src/client/components/CountrySelect/CountryList/CountryList.tsx]: any reason for passing down cycleUuid ? can the inner components get cycle with useCycle ?

**Review by minotogna — APPROVED** [review]

**Pull request merged**